.github/workflows/test.yml

```diff
@@ -0,0 +1,122 @@
+        #       Kubernetes versions with schemas available in:
+        #       https://github.com/instrumenta/kubernetes-json-schema
+        #
+        # NOTE: The "helm template" command will evaluate
+        #       .Capabilities.APIVersion.Has in templates based on a Kubernetes
+        #       version associated with the helm binary's version. Since we
+        #       render the templates with a specific helm version, we end up
+        #       rendering templates using a mocked k8s version unrelated to the
+        #       Kubernetes version we want to validate against. This issue has
+        #       made us not validate against versions lower than 1.14.
+        run: tools/templates/lint-and-validate.py --kubernetes-versions 1.14.0,1.18.0
+
+  test_z2jh:
+    runs-on: ubuntu-latest
+    # If a job goes on for this long it's probably hung
+    timeout-minutes: 20
+    strategy:
+      # Keep running so we can see if other tests pass
+      fail-fast: false
+      matrix:
+        include:
+          - k3s-version: v1.19.3+k3s1
+            helm-version: v3.4.0
+            test: install
+          - k3s-version: v1.18.10+k3s1
+            helm-version: v3.3.4
+            test: install
+          - k3s-version: v1.17.13+k3s1
+            helm-version: v3.3.4
+            test: install
+          - k3s-version: v1.16.15+k3s1
+            helm-version: v3.3.4
+            test: install
+
+          - k3s-version: v1.19.3+k3s1
+            helm-version: v3.3.4
+            test: upgrade
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # chartpress requires the full history
+          fetch-depth: 0
+      - uses: manics/action-k3s-helm@v0.2.1
+        with:
+          k3s-version: ${{ matrix.k3s-version }}
+          helm-version: ${{ matrix.helm-version }}
+          metrics-enabled: false
+          traefik-enabled: false
+          docker-enabled: true
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+      - name: Install and run chartpress
+        run: |
+          . ./ci/common
+          install_and_build
+      - name: Install latest JupyterHub version from Helm chart repo
+        if: matrix.test == 'upgrade'
+        run: |
+          . ./ci/common
+          helm install jupyterhub jupyterhub/jupyterhub --values dev-config.yaml
+          await_jupyterhub
+          await_autohttps_tls_cert_acquisition
+      - name: Install JupyterHub dev chart
+        run: |
+          . ./ci/common
+          helm upgrade --install jupyterhub ./jupyterhub --values dev-config.yaml
+          await_jupyterhub
+          await_autohttps_tls_cert_acquisition
+      - name: Wait 1 minute for everything to be ready
+        # TODO: find out exactly which resources are slow and wait for thoses
+        # So far errors have included:
+        # - some pods not being ready
+        # - network policies not being ready
+        run: sleep 1m
```

Comment:
Awww =/ I struggled so much to get the CI system run without these kinds of tiresome delays... I guess we can try debug it over time together though.

Reply:
I'd prefer it if we let it fail hard for a while, I really want this gone ^^. Perhaps also stop running tests when pushing to master to avoid too much errors triggering notifications etc?

Reply:
I hate it too 😢 

At the moment the tests take around 8 mins, compared to 7 without this. However as already mentioned one of the runs fails roughly 10% of the time, so we can expect a CI failure around 50% of the time. This means on average CI will take around 10-11 mins. In addition it takes maintainers 1-2 minutes to check the CI logs to see if it's a usual flakey failure or a genuine failure. Overall I think removing the delay increases the maintenance burden.

Reply:
Let's leave it there, I'm glad you are also motivated to get rid of it as well!

I believe I'm afraid of being alone in working to optimize this away if we don't let it make some noise, but it will indeed be a burden if it add noise as well.